pkg/metrics/metrics.go

```diff
@@ -7,17 +7,17 @@ import (
 
 // When adding metric names, see https://prometheus.io/docs/practices/naming/#metric-names
 const (
-	specialResourcesCreatedQuery = "sro_managed_resources_total"
+	createdSpecialResourcesQuery = "sro_managed_resources_total"
 	completedStatesQuery         = "sro_states_completed_info"
 	completedKindQuery           = "sro_kind_completed_info"
 )
 
 var (
```

Comment:
This is something I have a mixed feelings about.
Collectors could be changed into members of `metricsImpl` when `resources` pkg is ready for dependency injection (collector cannot be registered twice without prior unregistration). Even then, registration will be postponed until the object is created (right now it's during `init()`). Also, `New()` would need to return `error` to indicate registration failure (MustRegister causes panic in case of error)
Other solutions for current situation would a singleton.

Reply:
Indeed, this approach is very common when dealing with Prometheus metrics. I am not a fan of `init()` either, but I would also leave this as is until we hit a big problem here.